org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -107,6 +108,7 @@ public abstract class BaseDocumentLifeCycleHandler {
 	private WorkspaceJob validationTimer;
 	private WorkspaceJob publishDiagnosticsJob;
 	private Set<ICompilationUnit> toReconcile = new HashSet<>();
+	private Set<ICompilationUnit> toValidate = Collections.synchronizedSet(new HashSet<>());
```

Comment:
nit: not a big deal. Using `concurrentHashMap.newKeySet();` maybe a better choice.

> https://www.baeldung.com/java-concurrent-hashset-concurrenthashmap#thread-safe-hashset-using-collections-utility-class

Reply:
nice.